File: Python/misc/factorial.py
def factorial(num):
    fact = 1
    while num > 0:
        fact *= num
        num -= 1
    return fact

print(factorial(5))

def factorial_recursion(num):
    if(num == 0):
        return 1

    return num * factorial_recursion(num - 1)

print(factorial_recursion(15))

File: TypeScript/misc/factorial.ts
export function factorial(num: number): number {
    let fact = 1;
    while(num > 0) {
        fact *= num;
        num--;
    }
    return fact;
}

console.log(factorial(5))

export function factorialRecursion(num: number): number {
    if(num === 0) return 1;

    return num * factorialRecursion(num-1)
}

console.log(factorialRecursion(6))